browser/javascript/v2/react-app/src/ampli/index.js

```diff
@@ -0,0 +1,566 @@
+      return;
+    }
+
+    const env = options?.environment ?? 'development';
+    const apiKey = options?.client?.apiKey ?? ApiKey[env];
+
+    if (options?.client?.instance) {
+      this.amplitude = options?.client?.instance;
+    } else if (apiKey) {
+      this.amplitude = amplitude;
+      return this.amplitude?.init(apiKey, undefined, { ...DefaultOptions, ...options?.client?.options });
+    } else {
+      console.error("ERROR: ampli.load() requires 'environment', 'client.apiKey', or 'client.instance'");
+    }
+  }
+
+  /**
+   * Identify a user and set or update that user's properties.
+   *
+   * @param {string|undefined} userId The user's id.
+   * @param {Object} properties The user's properties.
+   * @param {string[]} [properties.optionalArray] Description for identify optionalArray
+   * @param {number} properties.requiredNumber Description for identify requiredNumber
+   * @param {IdentifyOptions} [options] Optional event options.
+   */
+  identify(userId, properties, options) {
+    if (!this.isInitializedAndEnabled()) {
+      return;
+    }
+
+    if (userId) {
+      options = Object.assign({}, options, { user_id: userId });
+    }
+
+    const ampIdentify = new amplitude.Identify();
+    if (properties != null) {
+      for (const [key, value] of Object.entries(properties)) {
+        ampIdentify.set(key, value);
+      }
+    }
+    return this.amplitude?.identify(
+      ampIdentify,
+      options,
+    );
+  }
+
+  /**
+   * Set Group for the current user
+   *
+   * @param {String} name
+   * @param {String|String[]} value
+   */
+  setGroup(name, value) {
+    if (!this.isInitializedAndEnabled()) {
+      return;
+    }
+
+    this.amplitude?.setGroup(name, value);
+  }
+
+  /**
+   * Identify a group and set or update that group's properties.
+   *
+   * @param {string} groupType The group type.
+   * @param {string|string[]} groupName The group name.
+   * @param {Object} properties The group's properties.
+   * @param {string} [properties.optionalString] Description for group optionalString
+   * @param {boolean} properties.requiredBoolean Description for group requiredBoolean
+   * @param {GroupOptions} [options] Options for this groupIdentify call.
+   */
+  groupIdentify(groupType, groupName, properties, options) {
+    if (!this.isInitializedAndEnabled()) {
+      return;
+    }
+
+    const amplitudeIdentify = new amplitude.Identify();
+    if (properties != null) {
+      for (const [key, value] of Object.entries(properties)) {
+        amplitudeIdentify.set(key, value);
+      }
+    }
+    return this.amplitude?.groupIdentify(groupType, groupName, amplitudeIdentify, options);
+  }
+
+  /**
+   * Track event
+   *
+   * @param {BaseEvent} event The event to track.
+   * @param {EventOptions} [options] Optional event options.
+   */
+  track(event, options) {
+    if (!this.isInitializedAndEnabled()) {
+      return;
+    }
+
+    return this.amplitude?.track(
+      event,
```

Comment:
Maybe the question for the browser core SDK. Since the event already includes event properties? What's the purpose of the second input event_properties?

Reply:
The first property can be a string. In this case, we still need the event_properties.

Reply:
`* @param {BaseEvent} event The event to track.`
The first input for ampli will be always a BaseEvent, will not be a string, right?

Reply:
You are right. I think it is a fair reason. I just checked the core SDK and in order to pass options property, we still need the second parameter. We can either keep the current implementation or `this.amplitude.track(event, undefined, options)`.

Reply:
Sounds good. Will change.